test/distributed/optim/test_apply_optimizer_in_backward.py

```diff
@@ -137,3 +140,21 @@ def test_multiple_optim_for_params(self) -> None:
             [model, model_with_opt_in_bwd],
             [opt_0, opt_1],
         )
+
+    def test_get_optimizers_in_backward(self):
+        # Create a simple test model
+        class TestModel(torch.nn.Module):
+            def __init__(self):
+                super(TestModel, self).__init__()
+                self.linear1 = torch.nn.Linear(10, 5)
+                self.linear2 = torch.nn.Linear(5, 2)
+
+        model = TestModel()
+
+        # Apply optimizers in backward
+        _apply_optimizer_in_backward(torch.optim.SGD, model.parameters(), {"lr": 0.01})
+        result = set(_get_in_backward_optimizers(model))
```

Comment:
How are optimizers hashed? (I am guessing just by object.)

nit: Maybe you should check that `len(_get_in_backward_optimizers(model))` is expected (i.e. equal to `len(list(model.parameters()))`)?